### PR TITLE
[JENKINS-63639] Update commons-io to 2.8.0

### PR DESCRIPTION
**Pull request by MRamonLeon**
See [JENKINS-63639](https://issues.jenkins-ci.org/browse/JENKINS-63639).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by rsandell — APPROVED** [review]

**Comment by daniel-beck** [comment]
Did you check the API for compatibility?

**Comment by MRamonLeon** [comment]
> Did you check the API for compatibility?

I checked the compatibility with https://abi-laboratory.pro/index.php?view=compat_report&lang=java&l=commons-io&v1=2.6&v2=2.7&obj=cd12c&kind=bin#Top The main risk I saw is a class which stopped implementing `Serializable`, it's `IOCase`. I searched it and I found some usages on very used plugins (like credentials), but it's not stored AFAICT: https://github.com/search?q=org%3Ajenkinsci+iocase&type=code There are more low exceptions but I didn't check everyone. The binary compatibility claims to be 96.9%. 

What's the expectation from this check? Do we have a threshold or so?

**Review by jeffret-b — APPROVED** [review]

**Comment by jvz** [comment]
@daniel-beck I checked for API compatibility in the release vote (my first on the Commons PMC!); they use japicmp to automate API comparison checks between versions: https://commons.apache.org/proper/commons-io/japicmp.html

**Review by jvz — APPROVED** [review]

File: bom/pom.xml [hunk 1]
```diff
@@ -66,7 +66,7 @@ THE SOFTWARE.
       <dependency>
         <groupId>commons-io</groupId>
         <artifactId>commons-io</artifactId>
-        <version>2.6</version>
+        <version>2.7</version>
```
Comment (jvz) [inline bom/pom.xml:69]:
```suggestion
        <version>2.8.0</version>
```

Why upgrade to an outdated release?

**Review by jvz — CHANGES_REQUESTED** [review]
Oops misread this.

File: bom/pom.xml (hunk 1 above)
Comment (MRamonLeon) [reply bom/pom.xml:69]: Because 2.8.0 was released yesterday, after my PR :angry: :man_facepalming:

Comment (MRamonLeon) [reply bom/pom.xml:69]: I see:
* Removed **_read_** from InfiniteCircularInputStreamtop
* Some serialVersionUID changes on serializable interfaces.

Not a big problem from my PoV

Comment (jvz) [reply bom/pom.xml]: LMAO I just noticed that after commenting as the announce email hadn't even been sent yet until like an hour after I said this!

**Review by jvz — APPROVED** [review]
❤️

File: bom/pom.xml (hunk 1 above)
Comment (MRamonLeon) [reply bom/pom.xml]: I just learnt a new acronym. Seriously, it's a parallel language :laughing:

**Review by timja — APPROVED** [review]

**Pull request merged**